Framework/DataHandling/inc/MantidDataHandling/SetSample.h

```diff
@@ -49,13 +50,18 @@ class MANTID_DATAHANDLING_DLL SetSample final : public API::Algorithm {
 
   const Geometry::SampleEnvironment *
   setSampleEnvironment(API::MatrixWorkspace_sptr &workspace,
-                       const Kernel::PropertyManager &args);
+                       const Kernel::PropertyManager_sptr &args);
```

Comment:
Passing the `shared_ptr`'s by const reference here just avoids increasing the reference count: http://stackoverflow.com/a/37610721. I know I have a reference to keep the object alive at the top level as this is all private. I would usually prefer a plain `const PropertyManager&` but it needs to be a `shared_ptr` to be fed into the child algorithm.  

You're right I can probably use `PropertyManager_const_sptr` here.